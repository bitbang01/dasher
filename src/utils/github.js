import qs from 'qs'

const GITHUB_CLIENT_ID = '799bd70f09884c025750'

export function getGithubAuthUrl({ origin }) {
  const redirectUri = origin + '/callback'

  const queryString = qs.stringify({
    client_id: GITHUB_CLIENT_ID,
    scope: 'repo',
    redirect_uri: redirectUri,
  })

  return `https://github.com/login/oauth/authorize?${queryString}`
}
