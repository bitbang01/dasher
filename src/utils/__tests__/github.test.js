import { getGithubAuthUrl } from '../github'

describe('getGithubAuthUrl()', () => {
  test('returns correct URL', () => {
    const location = {
      origin: 'http://localhost:3000',
    }

    expect(getGithubAuthUrl(location)).toMatchSnapshot()
  })
})
