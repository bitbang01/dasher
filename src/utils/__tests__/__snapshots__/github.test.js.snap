// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getGithubAuthUrl() returns correct URL 1`] = `"https://github.com/login/oauth/authorize?client_id=799bd70f09884c025750&scope=repo&redirect_uri=http%3A%2F%2Flocalhost%3A3000%2Fcallback"`;
